source/egg/math/Vector.cc

```diff
@@ -92,6 +92,18 @@ Vector3f Vector3f::minimize(const Vector3f &rhs) const {
     return out;
 }
 
+Vector3f Vector3f::proj(const Vector3f &rhs) const {
+    return rhs * rhs.dot(*this);
+}
+
+Vector3f Vector3f::rej(const Vector3f &rhs) const {
+    return *this - proj(rhs);
+}
+
+std::pair<Vector3f, Vector3f> Vector3f::projAndRej(const Vector3f &rhs) {
+    return std::pair(proj(rhs), rej(rhs));
+}
```

Comment:
We don't need this immediately... We will undoubtedly need it soon, probably for frame 3. Want to keep anyway?

Reply:
Sure, less diff noise later.